### PR TITLE
Implementation of additional proposed control operators and wasm bindings for validator

### LINKED ISSUE #82: `tstr` type allows maps?
Thank you for this fantastic implementation of CDDL! :sparkles: 

I've stumbled upon the following issue and wonder if it might be a bug?
The following CDDL schema (wrongly?) allows maps as values in the `fields` map even though only `tstr` types are permitted:

```
message = {
    fields: {
        + tstr => tstr
    }
}
```

This CBOR here (with diagnostic JSON) gets accepted by this schema while I would expect an error:

```
A1666669656C6473A16474657374A164546578746F48656C6C6F2C204D65737361676521
```

```json
{
  "fields": {
    "test": {
      "Text": "Hello, Message"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by anweiss**
Implements additional control operators as proposed in [draft-ietf-cbor-cddl-control-05](https://datatracker.ietf.org/doc/html/draft-ietf-cbor-cddl-control-05). CC @cabo

JSON:
- [x] numeric addition
- [x] string concatenation
- [x] string concatenation with dedenting
- [x] embedded abnf (ignores .abnfb)
- [x] features

CBOR:
- [x] numeric addition
- [x] string concatenation
- [x] string concatenation with dedenting
- [x] embedded abnf
- [x] features

Also includes wasm bindings for the validator module (#83), fixes for multiline byte strings and #88 and a rework of the CLI. Fixes #82 as well.

### CODE REVIEW

File: src/lexer.rs [hunk 1]
```diff
@@ -704,17 +704,33 @@ impl<'a> Lexer<'a> {
         '\x20'..='\x26' | '\x28'..='\x5b' | '\x5d'..='\u{10FFFD}' => {
           let _ = self.read_char();
         }
+        // SESC
+        '\\' => {
+          let _ = self.read_char();
+          if let Some(&(_, ch)) = self.peek_char() {
+            match ch {
+              '\x20'..='\x7e' | '\u{0128}'..='\u{10FFFD}' => {
```
Comment (cabo) [inline src/lexer.rs:712]: What is \u{0128}?  I decode that as Ĩ, which does not have a special meaning in CDDL.

Comment (anweiss) [reply src/lexer.rs:712]: @cabo And this is why I tagged you :) ... this is a typo ... thanks for catching! will fix

**Comment by cabo** [comment]
Is adding .abnf/.abnfb part of your plans?

**Comment by anweiss** [comment]
@cabo indeed it is

**Comment by cabo** [comment]
That is great to hear, because that will help us with moving forward on that internet-draft!

**Comment by cabo** [comment]
So is the non-ABNF part of this ready for prime time?
I was updating the "Implementation Status" section of draft-ietf-cbor-cddl-control.md and was about to point to your implementation...

**Comment by cabo** [comment]
> Since there don't seem to be any Rust crates that can validate a string against an ABNF grammar (there's https://crates.io/crates/abnf but it only parses ABNF), my current approach is to use https://crates.io/crates/abnf_to_pest to convert the ABNF to a [Pest](https://pest.rs/) grammar, and then validate the string against Pest per [pest-parser/pest#432](https://github.com/pest-parser/pest/issues/432). This isn't ideal, but the ABNF options in Rust are limited.

(That is pretty much the way I have been implementing ABNF: I have the ABNF to treetop translator in the `abnftt` gem, and use the treetop PEG parser then.  Note that this impresses a PEG interpretation on ABNF...)

**Comment by anweiss** [comment]
@cabo FYI, I've updated the description of the PR to denote which controls have been implemented as of today

**Comment by anweiss** [comment]
Moving this out of draft status. All control operators as of [draft-ietf-cbor-cddl-control-05](https://datatracker.ietf.org/doc/html/draft-ietf-cbor-cddl-control-05) have been implemented. Still missing some more unit tests before merging.

**Pull request merged**